### PR TITLE
Add class and unit test for JAVA-163

### PR DESCRIPTION
**Pull request by llinggit**
So we can incorporate your pull request, please share the following:
* What issue are you addressing with this pull request? - JAVA-163
* Are you modifying the correct branch? (See CONTRIBUTING.md) - develop
* Have you run unit tests? (See CONTRIBUTING.md) - yes
* Version of MarkLogic Java Client API (see Readme.txt) - 5.0
* Version of MarkLogic Server (see admin gui on port 8001) - ML 10
* Java version (`java -version`) - 11
* OS and version - Linux
* What Changed: What happened before this change? What happens without this change?

### CODE REVIEW

File: marklogic-client-api/src/test/java/com/marklogic/client/test/datamovement/XMLSplitterTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
+/*
```
Comment (ehennum) [inline marklogic-client-api/src/test/java/com/marklogic/client/test/datamovement/XMLSplitterTest.java]: Great unit test -- covers the core functionality.

File: marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java [hunk 2]
```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright 2015-2020 MarkLogic Corporation
```
[the remark this answers is not in the document]
Comment (llinggit) [reply marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java]: Changed java classes and unit test copy right to 2020. Test data doesn't have copyright currently.  Should I add copyright to test data as well?

File: marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+         * @param nsUri Namespace URI
+         * @param localName Element local name
+         */
+        public BasicElementVisitor(String nsUri, String localName) {
+            if (nsUri == null) {
+                nsUri = "";
+            }
+
+            if (localName == null) {
+                localName = "";
```
[the remark this answers is not in the document]
Comment (llinggit) [reply marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java]: Changed to
if (localName == null || localName == "") {
                throw new IllegalArgumentException("LocalName cannot be null");
}

File: marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+
+                            case PROCESS:
+                                T handle = visitor.makeBufferedHandle(xmlStreamReader);
+                                if (handle != null) {
+                                    return handle;
+                                }
+                                break;
+
+                            case SKIP:
+                                int count = 0;
```
[the remark this answers is not in the document]
Comment (llinggit) [reply marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java]: Yes. Changed "count" to "depth".

File: marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ehennum) [reply marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java]: Good catch.  Test sources also need the copyright comment.

[the remark this answers is not in the document]
Comment (llinggit) [reply marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java]: Added copyright to test data.

File: marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+                                while (xmlStreamReader.hasNext()) {
+                                    int next = xmlStreamReader.next();
```
[the remark this answers is not in the document]
Comment (llinggit) [reply marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java]: Added break label skipCheck to switch statement. No need of skipElement label.

File: marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+            setXmlStreamReader(input);
+            visitor = splitter.getVisitor();
+        }
+
+        T getNextHandle() {
+            try {
+                while (xmlStreamReader.hasNext()) {
+                    int event = xmlStreamReader.next();
+
+                    if (event == XMLStreamReader.START_ELEMENT) {
```
[the remark this answers is not in the document]
Comment (llinggit) [reply marklogic-client-api/src/main/java/com/marklogic/client/datamovement/XMLSplitter.java]: Changed if-else statement to switch statement. Added label checkForHandle for switch statement. No need of label whileSeekingHandle for while statement.

File: marklogic-client-api/src/test/java/com/marklogic/client/test/datamovement/XMLSplitterTest.java (hunk 1 above)
Comment (llinggit) [reply marklogic-client-api/src/test/java/com/marklogic/client/test/datamovement/XMLSplitterTest.java]: Thanks :)

**Pull request merged**